outlines/exploring-graphiql-the-in-browser-ide-for-graphql.md

```diff
@@ -0,0 +1,187 @@
+
+```graphql
+# add a simple code that references the type-ahead feature
+```
+
+[Screenshot a live view of type-ahead feature in action]
+
+B. Live syntax and validation error highlighting.
+
+```graphql
+"""
+add a simple code that references the live syntax and validation error
+"""
+```
+
+[Screenshot a live view of validation error highlighting]
+
+C. Query history
+
+[Screenshot the history interface]
+
+D. Fragments
+
+```graphql
+# code that references fragments that can be merged to a query
+```
+
+(Add a sample diagram showing the fragment part of the editor and how it is analysed)
+
+E. Documentation: GraphiQL generates documentation for your schema, making it easy to understand and explore.
+
+[Screenshot the documentation interface]
+
+## Testing out the Interface
+
+A. Writing the query
+
+B. Integrating Variables with Your Query
+
+C. Utilising fragments to aid query construction
+
+D. Executing/Testing the query: Sending requests and handling responses
+
+[take a screenshot and add code for each of the step above]
+
+## Debugging GraphQL Queries with GraphiQL
+
+### Practical approach to debugging
+
+A. Setting up your GraphQL endpoint in the GraphiQl (add a link to this section in the exploring GraphiQL article). Launch GraphiQl and configure the endpoint.
+
+(add screenshot and codebase for setting up the GraphQl endpoint)
+
+B. Write your query in the Query editor pane
+
+```graphql
+# your query
+```
+
+C. Check for Syntax errors: GraphiQL will automatically validate your query and highlight any syntax errors in red as you type
+Example: missing curly braces
+
+```graphql
+# code sample with this error
+```
+
+(Screenshot of the error interface)
+
+D. Inspect your Schema: Use the Documentation Explorer pane to understand your schema's available types, fields, and arguments. Click on the types and fields to view their details
+
+(screenshot each of this interface)
+
+E. Test with variables: Add variables in the Variables Editor to make your query dynamic.
+
+```graphql
+# query showing the variable added
+```
+
+(screenshot the variable pane)
+
+F. Analysing the response
+
+- Inspect the response in the Result Pane to ensure you're getting the expected data
+
+```json
+// sample of successful response
+```
+
+- Explain an instance where there is an error and how users can use the error messages to identify the issues.
+
+```json
+// sample of an error response
+```
+
+### Debugging Strategies
+
+A. Errors that can lead to bugs in the Queries
+
+- Syntax
+
+(Add a screenshot of a query in the editor with a syntax error)
+
+- Type Mismatches
+
+(Add a screenshot of a query in the editor with a type mismatch error)
+
+- Field Resolution issues.
+
+(Add a diagram analysing the field resolution issue)
+
+B. Error Messages
+
+- Understanding and interpreting GraphQL error messages
+
+C. Network Tab in Developer Tools
+
+- Analyzing network requests and responses for additional insights.
+
+### Best Practices
```

Comment:
Not relavant

Reply:
Alright, I'd remove the whole thing.

Reply:
> Not relavant

Here, I removed all the sections from "Key Features and Capabilities of GraphiQl" to "Debugging Strategies"